# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1298531257 18000
#      Thu Feb 24 02:07:37 2011 -0500
# Node ID 26d3164fd58d8c1f69eebb7d48c2dbfe33593595
# Parent  95b1c64c287f751eab796659e8a11d5b727d59c9
strchr.m: avoid implict string to number conversion

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,13 @@
+2011-02-24  John W. Eaton  <jwe@octave.org>
+
+	* strings/strchr.m: Avoid implicit string to number conversion.
+	Bug #32546.  From Kim Hansen <kim@i9.dk>.
+
 2011-02-22  Ben Abbott  <bpabbott@mac.com>
 
 	* plot/subplot.m: Use new looseinset property introduced by changeset
 	12467. Decrease spacing between subplots rows / columns. Clean up and
 	add comments.
 
 2010-02-22  Rik  <octave@nomad.inbox5.com>
 
diff --git a/scripts/strings/strchr.m b/scripts/strings/strchr.m
--- a/scripts/strings/strchr.m
+++ b/scripts/strings/strchr.m
@@ -41,17 +41,17 @@ function varargout = strchr (str, chars,
     mask = str == chars(1);
     for i = 2:length (chars)
       mask |= str == chars(i);
     endfor
   else
     ## Index the str into a mask of valid values.  This is slower than
     ## it could be because of the +1 issue.
     f = false (1, 256);
-    f(chars + 1) = true;
+    f(uint8(chars)+1) = true;
     ## Default goes via double -- unnecessarily long.
     si = uint32 (str);
     ## in-place
     ++si;
     mask = reshape (f(si), size (str));
   endif
   varargout = cell (1, nargout);
   varargout{1} = [];
